nautobot/dcim/filters.py

```diff
@@ -184,10 +184,11 @@ class SiteFilterSet(NautobotFilterSet, TenancyFilterSet, StatusModelFilterSetMix
         field_name="circuit_terminations",
         label="Has circuit terminations",
     )
-    devices = NaturalKeyOrPKMultipleChoiceFilter(
+    # TODO: change this filter to NaturalKeyOrPKMultipleChoiceFilter once
+    # https://github.com/nautobot/nautobot/issues/2875 is resolved.
+    devices = django_filters.ModelMultipleChoiceFilter(
         queryset=Device.objects.all(),
-        to_field_name="name",
-        label="Devices (name or ID)",
+        label="Devices (ID)",
     )
```

Comment:
A lot of these are arguably a loss of functionality, since `device.name`, `rack.name`, etc. are "locally unique" fields and it's probably pretty useful to be able to filter on those even if they're not "true" natural keys. I get that this change is more correct from a code perspective but I'm still a bit unhappy about it.

Reply:
Yeah I understand, let me ask the team for their opinion on this.

Reply:
Let's keep the semi-incorrect behavior in for now with the `TODO`. We're going to be cleaning up natural keys in 2.0 along with the fact that 2804 not going in until 2.0 and there would be a strong desire to fix #2875 in 2.0 anyway.

Reply:
Thrived with what @bryanculver said. This isn't going to be finalized so for now just keeping it working while we keep moving along towards the fix in #2875 is the right path forward.